### PR TITLE
Changelings cannot last resolt while being absorbed and can hear changeling comms.

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Allows changelings a 70% chance to hear messages on changeling comms if they don't have the ability.
Changelings cannot last resort while being absorbed.

## Why It's Good For The Game

Changeling hivemind is a useless ability that you buy only for other people not to hear you. Now at least they can hear you, although others still cannot talk to you.
Additionally absorbing other changelings is more viable now, as they cannot last resort burst if you are absorbing them.

## Changelog
:cl:
tweak: Changelings have a 70% chance to hear hivemind messages even if they do not have the ability.
tweak: Last resort can no longer be used while being absorbed.
/:cl:

### CODE REVIEW

**Comment by r1ks-iwnl** [comment]
Doubt this will fix much since the other cling can just use last resort while being grabbed.

**Comment by zeskorion** [comment]
Last resolt

**Comment by OceanFish1** [comment]
Last resort more like first resort

**Comment by eeSPee** [comment]
Bring back absorb another ling objective?

File: code/datums/saymode.dm [hunk 1]
```diff
@@ -33,8 +33,11 @@
 							if(LINGHIVE_LINK)
 								to_chat(M, msg)
 							if(LINGHIVE_OUTSIDER)
-								if(prob(40))
-									to_chat(M, "<span class='changeling'>We can faintly sense an outsider trying to communicate through the hivemind...</span>")
+								var/mob/living/L = M
+								if (!HAS_TRAIT(L, CHANGELING_HIVEMIND_MUTE) && prob(70))
+									to_chat(M, msg)
+								else
+									to_chat(M, "<span class='changeling'>You hear a faint chittering from within your mind...</span>")
```
Comment (MCterra10) [inline code/datums/saymode.dm]: "We"
conform to the changeling lore.

also at line 67

File: code/modules/antagonists/changeling/powers/headcrab.dm [hunk 2]
```diff
@@ -11,6 +11,14 @@
 
 /datum/action/changeling/headcrab/sting_action(mob/user)
 	set waitfor = FALSE
+	if(isliving(user))
+		var/mob/living/L = user
+		var/mob/living/puller = L.pulledby
+		if(puller)
+			var/datum/antagonist/changeling/other_ling = is_changeling(puller)
+			if(other_ling?.isabsorbing)
+				to_chat(user, "<span class='warning'>Your last resort is being disrupted by another changeling!</span>")
```
Comment (MCterra10) [inline code/modules/antagonists/changeling/powers/headcrab.dm]: "Our"
changeling lore

File: code/datums/saymode.dm (hunk 1 above)
Comment (PowerfulBacon) [inline code/datums/saymode.dm]:
```suggestion
									to_chat(M, "<span class='changeling'>We hear a faint chittering from within our mind...</span>")
```

File: code/datums/saymode.dm [hunk 3]
```diff
@@ -57,8 +60,11 @@
 								if (!HAS_TRAIT(L, CHANGELING_HIVEMIND_MUTE))
 									to_chat(M, msg)
 							if(LINGHIVE_OUTSIDER)
-								if(prob(40))
-									to_chat(M, "<span class='changeling'>We can faintly sense another of our kind trying to communicate through the hivemind...</span>")
+								var/mob/living/L = M
+								if (!HAS_TRAIT(L, CHANGELING_HIVEMIND_MUTE) && prob(70))
+									to_chat(M, msg)
+								else
+									to_chat(M, "<span class='changeling'>You hear a faint chittering from within your mind...</span>")
```
Comment (PowerfulBacon) [inline code/datums/saymode.dm]:
```suggestion
									to_chat(M, "<span class='changeling'>Wehear a faint chittering from within our mind...</span>")
```

File: code/modules/antagonists/changeling/powers/headcrab.dm (hunk 2 above)
Comment (PowerfulBacon) [inline code/modules/antagonists/changeling/powers/headcrab.dm]:
```suggestion
				to_chat(user, "<span class='warning'>Our last resort is being disrupted by another changeling!</span>")
```

**Comment by heepox** [comment]
making it 70% will just make me spam the same message like 7 times just to be shure.

File: code/datums/saymode.dm [hunk 4]
```diff
@@ -57,8 +60,11 @@
 								if (!HAS_TRAIT(L, CHANGELING_HIVEMIND_MUTE))
 									to_chat(M, msg)
 							if(LINGHIVE_OUTSIDER)
-								if(prob(40))
-									to_chat(M, "<span class='changeling'>We can faintly sense another of our kind trying to communicate through the hivemind...</span>")
+								var/mob/living/L = M
+								if (!HAS_TRAIT(L, CHANGELING_HIVEMIND_MUTE) && prob(70))
+									to_chat(M, msg)
+								else
+									to_chat(M, "<span class='changeling'>Wehear a faint chittering from within our mind...</span>")
```
Comment (MCterra10) [inline code/datums/saymode.dm]: Typo 😠 
```suggestion
									to_chat(M, "<span class='changeling'>We hear a faint chittering from within our mind...</span>")
```

**Comment by eeSPee** [comment]
> making it 70% will just make me spam the same message like 7 times just to be shure.

Agree. Maybe 70% is a tad too reliable.

**Review by utilbird — APPROVED** [review]
70% is fine. You wouldn't absorb someone for the sole reason of seeing their hivemind comms

**Pull request merged**